### PR TITLE
finish testing for ticket 5

### PR DESCRIPTION
**Pull request by gosamuel**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -85,10 +85,47 @@ describe("GET /api/articles", () => {
...
+        });
+      });
+  });
+  it("should have seven properties on each object", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((res) => {
+        res.body.result.forEach((article) =>
+          expect(Object.keys(article).length).toBe(7)
```
Comment (two-triangles) [inline __tests__/app.test.js]: We need to be checking in our tests what these properties actually are!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -85,10 +85,47 @@ describe("GET /api/articles", () => {
...
+  });
+  it("should return articles in desc date order", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((res) => {
+        res.body.result.forEach((article, index) => {
+          if (index < res.body.result.length - 1) {
+            expect(
+              article.created_at > res.body.result[index + 1].created_at
```
Comment (two-triangles) [inline __tests__/app.test.js]: You can use the jest-sorted library to do this in a much clearer way - have a go at implementing this

File: __tests__/app.test.js [hunk 3]
```diff
@@ -85,10 +86,46 @@ describe("GET /api/articles", () => {
...
+        expect(res.body.result).toBeSorted({ descending: true });
+      });
+  });
+  it("should have all properties correct", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((res) => {
+        res.body.result.forEach(
+          ((article) => expect(article).toHaveProperty("article_id"),
```
Comment (two-triangles) [inline __tests__/app.test.js]: I don't think this is working like you are intending it to. forEach can only take a single callback as an argument. You are passing in 6 of them, so anything after the first callback function is going to be ignored. 

Have a look at the MDN docs for forEach and see if you can refactor to the correct syntax

File: __tests__/app.test.js [hunk 4]
```diff
@@ -101,22 +102,21 @@ describe("GET /api/articles", () => {
...
       .expect(200)
       .then((res) => {
-        res.body.result.forEach((article, index) => {
-          if (index < res.body.result.length - 1) {
-            expect(
-              article.created_at > res.body.result[index + 1].created_at
-            ).toBe(true);
-          }
-        });
+        expect(res.body.result).toBeSorted({ descending: true });
```
Comment (HannahBurks) [inline __tests__/app.test.js]: We want to be specific here which is descending (we want this to be created_at - have a look at the jest sorted docs and how to give another argument of which bit you want to sorted

**Review by HannahBurks — CHANGES_REQUESTED** [review]
Almost there Sam, just another tweak to your sorting

**Review by cjplatten — COMMENTED** [review]
Looks good!

**Pull request merged**